src/mailroom.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+from __future__ import division
+
+
+from sys import exit
+
+
+try:
+    input = raw_input
+except NameError:
+    pass
+
+
+DONOR_DICT = {
+    u'baker': [10.0, 20.0],
+    u'williams': [30.0, 40.0, 50.0],
+    u'davidson': [35.0, 28.0, 60.0]
+}
+
+
+def initial_prompt():
+    """Run the main menu prompt."""
+    user_menu_choice = input(u'PRESS: (L) - Thank You Letter, (R) - Report, '
+                             '(Q) - Exit : ').lower()
+    return user_menu_choice
+
+
+def initial_prompt_validator(user_menu_choice):
+    """Validate user input from the initial_prompt."""
+    if user_menu_choice in (u'l', u'r'):
+        return user_menu_choice
+    elif user_menu_choice == 'q':
+        exit()
+    else:
+        print(u'Invalid Input')
+        return u'invalid input'
+
+
+def name_prompt():
+    """Prompt user for a name, or asks for list to print the names in dict."""
+    donor_name_input = input(u'Enter Donor Name ((L) - List of Donors '
+                             u'(Q) - return to Main Menu) : ').lower()
+    return donor_name_input
+
+
+def name_prompt_validator(donor_name_input):
+    """Validate name_prompt's input."""
+    if donor_name_input == u'q':
+        main_function()
+    else:
+        try:
+            if int(donor_name_input) or float(donor_name_input):
+                print(u'Invalid Input')
+                return u'invalid input'
+        except ValueError:
+            if donor_name_input != ' ':
+                return donor_name_input
+            else:
+                print(u'Invalid Input')
+                return u'invalid input'
+
+
+def amount_prompt():
+    """Prompt for donation amount."""
+    donation_amount = input(u'Enter Donation Amount  '
+                            u'((Q) - return to Main Menu) : ')
+    return donation_amount
+
+
+def amount_prompt_validator(donation_amount):
+    """Validate donation amount."""
+    if donation_amount == u'q':
+            main_function()
+    else:
+        try:
+            if int(donation_amount) or float(donation_amount):
+                return float(donation_amount)
+        except ValueError:
+            print(u'Invalid Input')
+            return u'invalid input'
+
+
+def update_dict(donor_name, donation_amount, dictionary):
+    """Check if name is in dict, add if not and add donation to the dict."""
+    dictionary.setdefault(donor_name, []).append(donation_amount)
+    return dictionary
```

Comment:
I'm not convinced this function has any value.  It's really just a one-line mutation of the passed dictionary.  I would just write this one line in-place in the code where the function is currently called.